.github/workflows/release.yml

```
@@ -21,12 +23,52 @@ permissions:
   contents: read
 
 jobs:
+  check:
+    name: Check which projects need to be built
+    runs-on: ubuntu-latest
+    outputs:
+      projects: ${{ steps.projects.outputs.projects }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - id: projects
+        env:
+          TARGET_REF: ${{ github.base_ref || github.ref }}
+        run: |
+          projects=()
+          for project in node miner; do
+            echo "project=$project"
+            current_version=$(go run cmd/release/main.go --json $project version | jq -r .msg)
+            echo "${project}_current_version=$current_version"
+            if [[ $current_version =~ -dev$ ]]; then
+              echo "Skipping $project because it's a dev version"
+              continue
+            fi
+            if [[ $current_version =~ -rc$ && "$TARGET_REF" == 'refs/heads/releases' ]]; then
+              echo "Skipping $project because it's an RC version on the releases branch"
+              continue
+            fi
+            if [[ "$project" == "node" ]]; then
+              current_version_exists=(git tag -l 'v*' | grep -q $current_version)
+            else
+              current_version_exists=(git tag -l '$project/v*' | grep -q $current_version)
+            fi
+            echo "${project}_current_version_exists=$current_version_exists"
+            if [[ ! $current_version_exists ]]; then
+              projects+=($project)
```

```suggestion
              projects+=($project)
              echo "Adding $project to build list"
```

Not sure how helpful this is ultimately but as long as we're debugging skip cases